src/metabase/models/params/custom_values.clj

```diff
@@ -49,28 +50,29 @@
   Maybe we should lower it for the sake of displaying a parameter dropdown."
   1000)
 
-
 (defn- values-from-card-query
-  [card value-field query]
-  (let [value-base-type (:base_type (qp.util/field->field-info value-field (:result_metadata card)))
-        expressions (get-in card [:dataset_query :query :expressions])]
+  [card value-field-ref query]
+  (let [value-base-type (:base_type (qp.util/field->field-info value-field-ref (:result_metadata card)))
+        value-field-ref (lib.util.match/replace value-field-ref
+                          [:expression expr-name opts]
+                          [:field expr-name (merge {:base-type value-base-type} opts)]
+
+                          [:expression expr-name]
+                          [:field expr-name {:base-type value-base-type}])]
     {:database (:database_id card)
      :type     :query
-     :query    (merge
-                 (cond-> {:source-table (format "card__%d" (:id card))
-                          :breakout     [value-field]
-                          :limit        *max-rows*}
-                   expressions
-                   (assoc :expressions expressions))
```

Comment:
no idea what this was supposed to be accomplishing but it lead to queries that had every expression repeated twice, which somehow used to still work when we nested every expression but stopped working when we stopped doing that nesting. Took it out and things are working again.